app/src/main/java/com/mathewsachin/fategrandautomata/imaging/DroidCvPattern.kt

```diff
@@ -12,19 +12,36 @@ import java.io.InputStream
 import kotlin.math.roundToInt
 import org.opencv.core.Size as CvSize
 
-class DroidCvPattern(private var Mat: Mat? = Mat(), private val OwnsMat: Boolean = true) :
-    IPattern {
-    constructor() : this(Mat())
+class DroidCvPattern(
+    private var Mat: Mat? = Mat(),
+    private val OwnsMat: Boolean = true
+) : IPattern {
+    private data class MatWithAlpha(val mat: Mat, val alpha: Mat?)
+
+    var alpha: Mat? = null
 
     private companion object {
-        fun makeMat(Stream: InputStream): Mat {
+        fun makeMat(Stream: InputStream): MatWithAlpha {
             val byteArray = Stream.readBytes()
             DisposableMat(MatOfByte(*byteArray)).use {
-                return Imgcodecs.imdecode(it.Mat, Imgcodecs.IMREAD_GRAYSCALE)
+                val decoded = Imgcodecs.imdecode(it.Mat, Imgcodecs.IMREAD_UNCHANGED)
+
+                // If there are 4 channels (RGBA), last one is alpha
+                val alphaChannel = if (decoded.channels() == 4) {
+                    Mat().apply { Core.extractChannel(decoded, this, 3) }
+                } else null
+
+                Imgproc.cvtColor(decoded, decoded, Imgproc.COLOR_RGBA2GRAY)
+
+                return MatWithAlpha(decoded, alphaChannel)
             }
         }
     }
 
+    private constructor(matWithAlpha: MatWithAlpha) : this(matWithAlpha.mat) {
```

Comment:
Is there any situation where we would have a constructor call with MatWithAlpha and ownsMat = false? ownsMat is part of the default constructor.

Reply:
`ownsMat` is only used for screenshots where alpha doesn't matter